internal_transfers/actions/src/parse.ts

```diff
@@ -23,32 +20,42 @@ export function partitionEventLogs(logs: Log[]): ClassifiedEvents {
     trades: [],
     settlements: [],
   };
+  const settlementInterface = new ethers.Interface(SETTLEMENT_ABI);
+  const erc20Interface = new ethers.Interface(IERC20_ABI);
   logs.forEach((log, index) => {
-    const topics = log.topics;
-    const eventTopic = topics[0];
-
-    if (eventTopic === TRANSFER_EVENT_TOPIC) {
+    // We are only interested in Transfer Events from erc20 contracts
+    // along with Settlement and Trade Events from the Settlement contract
+    // All other event emissions can be ignored for our purposes.
+    const transferEventLog = erc20Interface.parseLog(log);
+    const settlementEventLog = settlementInterface.parseLog(log);
+    if (transferEventLog != null) {
+      const { from, to } = transferEventLog.args;
       const transfer: TransferEvent = {
-        to: addressFromBytes32(topics[2]),
-        from: addressFromBytes32(topics[1]),
-        amount: BigInt(log?.data),
+        from: from,
+        to: to,
+        amount: BigInt(transferEventLog.args[2]),
       };
       // Is a "relevant" transfer (involving settlement contract)
       if (transferInvolves(transfer, SETTLEMENT_CONTRACT_ADDRESS)) {
         result.transfers.push(transfer);
       }
-    } else if (eventTopic === TRADE_EVENT_TOPIC) {
-      result.trades.push({
-        owner: addressFromBytes32(topics[1]),
-      });
-    } else if (eventTopic === SETTLEMENT_EVENT_TOPIC) {
-      result.settlements.push({
-        solver: addressFromBytes32(topics[1]),
-        log_index: index,
-      });
+    } else if (settlementEventLog != null) {
+      // Relevant Event Types
+      if (settlementEventLog.name == "Trade") {
```

Comment:
I think this string comparison would probably be better off as a topic hash comparison. although then this code is going to wind up looking almost exactly like what we had before... Anyway, I will change it to compare topic hash.... I guess it would still be possible to write a contract having an event with identical topic that is emitted during a settlement... so we probably also need to validate the emission address of the event when classifying as well.

Reply:
> it would still be possible to write a contract having an event with identical topic that is emitted during a settlement

The event decoding step by `ethers` is already internally computing the signature hash and only selecting events with that hash. As long as your ABIs don't contain events with repeating names (e.g., `Trade(uint256)` and `Trade(address)`) then there shouldn't be any name conflict issue.
It's still true that a token contract could emit an event that has the exact same hash as the settlement contract, so I'd check the emitter address as well anyway (but checking the event hashes here wouldn't help).

Reply:
Actually this is already the "settlementEventLog" so we know the emitter. I think this part is fine.